### PR TITLE
Feature/create dynamic form templates

### PR DESCRIPTION
**Pull request by akashP2312**

### CODE REVIEW

File: src/app/common/services/forms.service.ts [hunk 1]
```diff
@@ -1,26 +1,36 @@
...
 
   GetFormTemplates() {
-    return this.tempFormTemplatesStorage;
+    return this.http.get('http://localhost:8080/forms/');
   }
 
-  GetFormTemplate(formId: any) {
-    return this.tempFormTemplatesStorage.find((form: { Id: any; }) => form.Id === formId);
+  GetFormTemplate(formName: any): Observable<IGetFormTemplateResponse> {
+    return this.http.get<IGetFormTemplateResponse>(`http://localhost:8080/forms/${formName}/`);
```
Comment (cpandya231) [inline src/app/common/services/forms.service.ts]: Lets use base url from ServiceUtil.

File: src/app/common/services/forms.service.ts [hunk 2]
```diff
@@ -1,26 +1,36 @@
...
-    this.tempFormTemplatesStorage.push(formTemplate);
+  SaveFormTemplate(formTemplate: any, name: string, id: number) {
+    let body = {
+      name,
+      template: JSON.stringify({formName: name, components: formTemplate.components})
+    }
+    if (id) {
+      body = {...body, ...{id}}
+    }
+    return this.http.post('http://localhost:8080/forms/save', body,{
```
Comment (cpandya231) [inline src/app/common/services/forms.service.ts]: Lets use base url from ServiceUtil.

File: src/app/common/services/global-app-intercepter.service.ts [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+export class GlobalAppIntercepter implements HttpInterceptor {
+    constructor(private authService: AuthService) {}
+    intercept(req: HttpRequest<any>, next: HttpHandler): Observable<HttpEvent<any>> {
+        let authToken;
+        const commonHeaders = {
+            'Access-Control-Allow-Origin': '*',
+            'Access-Control-Allow-Methods': 'GET, POST, PATCH, PUT, DELETE, OPTIONS',
+            'Access-Control-Allow-Headers': 'Origin, Content-Type, X-Auth-Token'
+        };
+        if (req.url.indexOf('login') !== -1) {
```
Comment (cpandya231) [inline src/app/common/services/global-app-intercepter.service.ts:17]: Can we use switch case instead of if else?

**Review by cpandya231 — CHANGES_REQUESTED** [review]

**Comment by cpandya231** [comment]
We also need to resolve conflicts with main

File: src/app/common/services/global-app-intercepter.service.ts (hunk 3 above)
Comment (akashP2312) [reply src/app/common/services/global-app-intercepter.service.ts:17]: we can. but I think it doesn't make much difference here as we only have 2 conditions and we are not comparing with any fix value

File: src/app/common/services/forms.service.ts (hunk 2 above)
Comment (akashP2312) [reply src/app/common/services/forms.service.ts]: done

**Review by cpandya231 — APPROVED** [review]

**Pull request merged**